components/schema-migrator/migrations/director/20221206114754_add-tenant-to-formation-templates.up.sql

```diff
@@ -0,0 +1,15 @@
+BEGIN;
+
+ALTER TABLE formation_templates ADD COLUMN tenant_id uuid;
+
+ALTER TABLE formation_templates
+    ADD CONSTRAINT formation_templates_tenant_fk
+        FOREIGN KEY (tenant_id) REFERENCES business_tenant_mappings (id) ON DELETE CASCADE;
```

Comment:
Currently, tenant scoped formation templates has ON DELETE CASCADE when a tenant is deleted.
Formations also has ON DELETE CASCADE when a tenant is deleted.
However, formations also has a FK to formation templates which does NOT have ON DELETE CASCADE.

This creates and ambiguous trigger execution path during our tenant orphan data deletion.

Let's imagine that a tenant is deleted, if this first deletes cascadely the formations and then the formation templates everything is fine. 

However, if it first deletes the formation templates, and there are formations of these types, this will fail and the whole tenant deprovision will fail.